cedar-policy-core/src/ast/policy_set.rs

```diff
@@ -96,6 +118,50 @@ pub enum PolicySetError {
     },
 }
 
+/// Potential errors when working with `PolicySet`s.
+#[derive(Error, Debug)]
+pub enum PolicySetGetLinksError {
+    /// There was no [`PolicyID`] in the set of templates.
+    #[error("No template `{0}`")]
+    MissingTemplate(PolicyID),
+}
+
+/// Potential errors when unlinking from a `PolicySet`.
+#[derive(Error, Debug)]
+pub enum PolicySetUnlinkError {
+    /// There was no [`PolicyID`] linked policy to unlink
+    #[error("unable to unlink policy id `{0}` because it does not exist")]
+    UnlinkingError(PolicyID),
+}
+
+/// Potential errors when removing templates from a `PolicySet`.
+#[derive(Error, Debug)]
+pub enum PolicySetTemplateRemovalError {
+    /// There was no [`PolicyID`] template in the list of templates.
+    #[error(
+        "unable to remove static policy id `{0}` from template list because it does not exist"
+    )]
+    RemovePolicyNoTemplateError(PolicyID),
```

Comment:
unable to remove a static policy, or unable to remove a template? the error message and error variant name don't seem to match